gitpod-io/gitpod
# Streamline code formatting

## Description

## Why

- I'm about to start working on fixing the linting problems the dashboard app logs, but ESLint and Prettier have some common gray area and it's better to start with consistently formatted code to have an easier time in handling this common area
- Basically why Prettier exists: having an automated coding style which is readable, consistent, and not particularly prone to be argued upon: again, @loujaybee this improves DevX

## What

- Adds Prettier support and configurations
- Format on save
- Batch format everything Prettier can format to obtain a good starting point
- Monorepo components / folders are batch-formatted in separate commits in case you prefer to accept this only for some of them

## TODO

- [ ] Get feedback, is this useful? (I think it's fundamental!)
- [ ] Get feedback, do we want it everywhere is supported? (again, see the commits)
- [ ] Add notes to `README` (which ones depends on the answer to the question just above) about this
- [ ] Add a pre-commit hook, just in case. But this seems better done in a separate PR after / with ESLint because Prettier suggests doing it with https://github.com/okonet/lint-staged#configuration and `lint-staged` suggests stabilizing ESLint before configuring it
- [ ] Get someone to try it with a Jetbrains IDE

## How to test


Use, build, test Gitpod to see if this broke something (it shouldn't have, since Prettier only reformats code)
